src/Routing/Route.php

```diff
@@ -30,28 +30,28 @@ class Route
      * @param class-string<RequestHandlerInterface> $controller
      * @param Methods                               ...$methods
      */
-    public function __construct(string $uri, public readonly string $controller, Methods ...$methods)
+    public function __construct(protected readonly Router $router, string $uri, public readonly string $controller, Methods ...$methods)
```

Comment:
you should consider breaking down the arguments into newlines to comply with PSR12
eg:
```php
public function __construct(
        protected readonly Router $router,
        string $uri,
        public readonly string $controller,
        Methods ...$methods
    ) {
        // ...
```